python/cudf/cudf/tests/test_string.py

```diff
@@ -3423,3 +3423,37 @@ def test_str_join_lists(sr, sep, string_na_rep, sep_na_rep, expected):
         sep=sep, string_na_rep=string_na_rep, sep_na_rep=sep_na_rep
     )
     assert_eq(actual, expected)
+
+
+def test_str_find_multiple():
+    s = cudf.Series(["strings", "to", "search", "in"])
+    t = cudf.Series(["a", "s", "g", "i", "o", "r"])
+
+    expected = cudf.Series(
+        [
+            [-1, 0, 5, 3, -1, 2],
+            [-1, -1, -1, -1, 1, -1],
+            [2, 0, -1, -1, -1, 3],
+            [-1, -1, -1, 0, -1, -1],
+        ]
+    )
+
+    assert_eq(s.str.find_multiple(t).to_pandas(), expected.to_pandas())
+
+    s = cudf.Index(s)
+    t = cudf.Index(t)
+
+    expected.index = s
+
+    assert_eq(s.str.find_multiple(t).to_pandas(), expected.to_pandas())
+
+
+def test_str_find_multiple_error():
+    s = cudf.Series(["strings", "to", "search", "in"])
+    with pytest.raises(
+        TypeError,
+        match=re.escape(
+            "patterns can only be a Series/Index, got: <class 'str'>"
+        ),
+    ):
+        s.str.find_multiple("a")
```

Comment:
What error is raised when a Series/Index of non-strings is passed?

Reply:
Previously it was a `RuntimeError` like below:


```python
In [1]: import cudf

In [2]: s = cudf.Series(["strings", "to", "search", "in"])

In [3]: t = cudf.Series([1, 2, 3])

In [4]: s.str.find_multiple(t)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In [4], line 1
----> 1 s.str.find_multiple(t)

File /nvme/0/pgali/envs/cudfdev/lib/python3.9/site-packages/cudf/core/column/string.py:3667, in StringMethods.find_multiple(self, patterns)
   3660 if not isinstance(patterns, (cudf.Series, cudf.Index)):
   3661     raise TypeError(
   3662         "patterns can only be a Series/Index, "
   3663         f"got: {type(patterns)}"
   3664     )
   3666 return cudf.Series(
-> 3667     libstrings.find_multiple(self._column, patterns._column),
   3668     index=self._parent.index
   3669     if isinstance(self._parent, cudf.Series)
   3670     else self._parent,
   3671     name=self._parent.name,
   3672 )

File find_multiple.pyx:24, in cudf._lib.strings.find_multiple.find_multiple()

RuntimeError: cuDF failure at: /nvme/0/pgali/cudf/cpp/src/strings/strings_column_view.cpp:24: strings_column_view only supports strings
```

But now changed it to a much cleaner `TypeError`:
```python
TypeError: patterns can only be of 'string' dtype, got: int64
```